### PR TITLE
[pg] Add missing parameter to BindConfig

### PR DESCRIPTION
**Pull request by iamkhush**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [url here](https://github.com/brianc/node-postgres/blob/master/packages/pg-protocol/src/serializer.ts#L112)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/pg/index.d.ts [hunk 1]
```diff
@@ -118,11 +118,14 @@ export interface QueryParse {
     types: string[];
 }
 
+type ValueMapper = (param: any, index: number) => any;
+
 export interface BindConfig {
-    portal?: string | undefined;
```
Comment (jakebailey) [inline types/pg/index.d.ts:122]: I highly suspect that these should all retain `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

I would recommend not changing this and keeping the PR minimal.

Comment (iamkhush) [reply types/pg/index.d.ts:122]: Makes sense. 
I blindly copied it from 
https://github.com/brianc/node-postgres/blob/master/packages/pg-protocol/src/serializer.ts#L106
to make those consistent  . I will revert changes in non-required places

Comment (jakebailey) [reply types/pg/index.d.ts:122]: Oh, huh. Why doesn't this package ship its types if it's written in typescript?

Comment (jakebailey) [reply types/pg/index.d.ts:122]: Ah, `pg` isn't typed. But, `pg-protocol` is, so should `@types/pg` just pull that in?

**Review by jakebailey — APPROVED** [review]

File: types/pg/index.d.ts (hunk 1 above)
Comment (iamkhush) [reply types/pg/index.d.ts:122]: But then it would break things , like when I blindly updated the types before.

Comment (jakebailey) [reply types/pg/index.d.ts:122]: Not wrong, but, then it's probably on the upstream repo to fix.

Comment (iamkhush) [reply types/pg/index.d.ts:122]: What do you mean ? Should I ask the maintainers of the pg package to get involved ?

Comment (jakebailey) [reply types/pg/index.d.ts:122]: I just mean that if the types weren't duplicated here and instead used the actual dependencies of the `pg` package, that any issue relating to the missing `| undefined` would be reported upstream and we probably wouldn't be having this conversation. There's nothing really to do here.

**Comment by iamkhush** [comment]
@jakebailey @pspeter3 @HoldYourWaffle Could I get a merge here please ?

**Review by jakebailey — APPROVED** [review]

**Comment by iamkhush** [comment]
Ready to merge

**Pull request merged**